api/database/sampleApp.py

```diff
@@ -0,0 +1,64 @@
+from flask import Flask, request, jsonify
+from mongoengine import connect, disconnect
+from pymongo.errors import DuplicateKeyError
+from users import Users
+
+app = Flask(__name__)
+
+@app.route('/users', methods = ["GET"])
+def get_users():
+    name = request.args.get('Name')
+    email = request.args.get('Email')
+    password = request.args.get('Password')
+
+    data = []
+    for user in Users.objects:
+        if name is not None:
+            if user.name == name:
+                data.append(user.to_json())
+        elif email is not None:
+            if user.email == email:
+                data.append(user.to_json())
+        else:
+            data.append(user.to_json())
+    
+    dataDict = {'users': data}
+    
+    return dataDict, 200
```

Comment:
Either remove this `get_users` route, or rewrite it in the style of `get_users_name`, since that uses the `Users.objects(name = name ... etc)` , that will make things much cleaner and make the example a lot less confusing I think.